tests/mqtt_mock_server_handler.h

```diff
@@ -407,3 +467,152 @@ static void s_mqtt_mock_server_wait_for_pubacks(struct aws_channel_handler *hand
     aws_condition_variable_wait_pred(&testing_handler->cvar, &testing_handler->lock, s_is_pubacks_complete, &waiter);
     aws_mutex_unlock(&testing_handler->lock);
 }
+
+/* this is only safe to call when not attached to a channel. */
+struct aws_array_list *mqtt_mock_server_get_received_messages(struct aws_channel_handler *handler) {
+    struct mqtt_mock_server_handler *testing_handler = handler->impl;
+
+    return &testing_handler->received_messages;
+}
+
+/* dynamic append from to to, the cursor of the buffer we just appended stored in result */
+static int s_move_cursor_to_buf_dynamic(
+    struct aws_byte_buf *to,
+    struct aws_byte_cursor *from,
+    struct aws_byte_cursor *result) {
+
+    size_t pre_len = to->len;
+    (void)pre_len;
+    /* move the cursor to buffer */
+    ASSERT_SUCCESS(aws_byte_buf_append_dynamic(to, from));
+    *result = aws_byte_cursor_from_buf(to);
+    /* move the result cursor to point at the place we just moved to */
+    aws_byte_cursor_advance(result, pre_len);
+
+    return AWS_OP_SUCCESS;
+}
+
+size_t mqtt_decoded_packets_count(struct aws_channel_handler *handler) {
```

Comment:
A number of the following top-level utility functions are missing the mqtt_mock_server_ prefix